dlt/common/destination/reference.py

```diff
@@ -560,14 +562,19 @@ def with_staging_dataset(self) -> ContextManager["JobClientBase"]:
 class SupportsStagingDestination:
     """Adds capability to support a staging destination for the load"""
 
+    def __init__(self, truncate_table_before_load_on_staging_destination: bool = True) -> None:
```

Comment:
I think you can take the full DestinationClientDwhWithStagingConfiguration as argument here.

Reply:
And it should be required, no default value. That way we can ensure that it is implemented correctly in each destination.